Always show "Select Primary Disk" screen (fixes xenserver#7)

ydirson opened the pull request:
Not touching this branch as it is also the source of xenserver#29.

Merge is available in branch `next` as 2a1091558b845f88a9d9c834cf6fd427da549547.

Wescoeur commented:
Seems ok! Thank you for your contrib! :)

bdonnahue commented:
@Wescoeur any idea how to get this baked into the 8.0.0 beta ISO file being distributed?

stormi commented:
It will be in the 8.0.0 final ISO. But if you want to do tests of net-install with answerfile before that, I can make one for you to test, I'm interested in the results.

bdonnahue commented:
@stormi appreciate the offer! but no worries, I have injected the code into my own ISO locally. Prior to submitting the PR I also did some tests using the answerfile and the answerfile_generator and all seemed AOK. The test scope was fairly limited though (mainly to the features etc I was using). Let me know if you'd like to see those answer files / generator files.

stormi commented:
Yes, sharing your test procedures could be useful both for us and potentially for other users.

stormi commented on answerfile.py:
```diff
@@ -90,9 +90,10 @@ def processAnswerfileSetup(self):
 
     def processAnswerfile(self):
         xelogging.log("Processing XML answerfile for %s." % self.operation)
+        results = {}
```
I'm pretty sure that this is not needed since the results variable is initialized between lines 97 and 102 (unless the install type is not recognized, but then an exception is raised).

bdonnahue replied:
@stormi I believe you're right. I was misunderstanding python scope as mentioned here: https://stackoverflow.com/questions/2829528/whats-the-scope-of-a-variable-initialized-in-an-if-statement

Line 93 could be deleted I believe

stormi commented:
I made new ISOs meant to fix the unattended installation (either remote upgrade method or installation with answer file). That's the only change so I've not made it an official RC2, just some tests ISOs for the community to confirm that the bug is fixed, so that we are sure that it will not be in the final ISOs.

https://updates.xcp-ng.org/tmp/RC1custom1/

ydirson committed:
Always show "Select Primary Disk" screen (fixes xenserver#7)

Skipping the primary disk slection when there is only one caused
several problems:

- hitting "Back" from "Virtual Machine Storage" screen tries to get
  back to primary disk selection, and finally has no effect
- users are sometimes confusing the disk selection for VM storage
  with the selection of primary disk

Signed-off-by: Yann Dirson <yann.dirson@vates.fr>
